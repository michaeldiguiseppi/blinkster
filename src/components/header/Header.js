import React, { Component } from 'react';

export default class Header extends Component {
    render() {
        let { title } = this.props;
        return (
            <nav className="navbar navbar-expand-lg navbar-dark bg-primary">
                <a className="navbar-brand" href="/">{ title }</a>
            </nav>
        )
    }
}