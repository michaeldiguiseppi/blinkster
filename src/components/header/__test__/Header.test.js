import React from 'react';
import Header from '../Header';
import renderer from 'react-test-renderer';
import { shallow, configure, mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

configure({ adapter: new Adapter() });


describe('<Header /> component', () => {
    const defaultProps = {
        
    }

    it('should render without breaking', () => {
        const headerComponent = mount(<Header {...defaultProps} />);
        expect(headerComponent.exists()).toBe(true);
        headerComponent.unmount();
    });

    it('should match the snapshot', () => {
        const tree = renderer
            .create(<Header {...defaultProps} />)
            .toJSON();
        expect(tree).toMatchSnapshot();
    });
});