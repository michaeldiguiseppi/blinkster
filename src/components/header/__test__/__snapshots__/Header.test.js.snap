// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> component should match the snapshot 1`] = `
<nav
  className="navbar navbar-expand-lg navbar-dark bg-primary"
>
  <a
    className="navbar-brand"
    href="#"
  />
  <button
    aria-controls="navbarColor01"
    aria-expanded="false"
    aria-label="Toggle navigation"
    className="navbar-toggler"
    data-target="#navbarColor01"
    data-toggle="collapse"
    type="button"
  >
    <span
      className="navbar-toggler-icon"
    />
  </button>
  <div
    className="collapse navbar-collapse"
    id="navbarColor01"
  >
    <ul
      className="navbar-nav mr-auto"
    >
      <li
        className="nav-item active"
      >
        <a
          className="nav-link"
          href="#"
        >
          Home
        </a>
      </li>
    </ul>
    <form
      className="form-inline my-2 my-lg-0"
    >
      <input
        className="form-control mr-sm-2"
        placeholder="Search"
        type="text"
      />
      <button
        className="btn btn-secondary my-2 my-sm-0"
        type="submit"
      >
        Search
      </button>
    </form>
  </div>
</nav>
`;
